components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -229,47 +229,43 @@ func (g *rtkSerial) start() error {
 
 // getStream attempts to connect to ntrip stream. We give up after maxAttempts unsuccessful tries.
 func (g *rtkSerial) getStream(mountPoint string, maxAttempts int) error {
-	g.mu.Lock()
-	defer g.mu.Unlock()
+	success := false
+	attempts := 0
 
 	var rc io.ReadCloser
 	var err error
 
 	g.logger.Debug("Getting NTRIP stream")
 
-	for attempts := 0; attempts < maxAttempts; attempts++ {
-		if g.isClosed {
-			return g.err.Get()
-		}
-
+	for !success && attempts < maxAttempts {
 		select {
 		case <-g.cancelCtx.Done():
 			return errors.New("Canceled")
 		default:
 		}
 
 		rc, err = func() (io.ReadCloser, error) {
+			g.mu.Lock()
+			defer g.mu.Unlock()
 			return g.ntripClient.Client.GetStream(mountPoint)
```

Comment:
This line is still doing something slow while we have the mutex locked. We're not obviously interacting with any state in this struct, so it's not clear to me why the mutex is important here. Could we get rid of lines 248-249 entirely?